File: java how to program-PersonalSolution/Exercises/8.4/Rectangle.java
public class Rectangle {


       private double length;
       private double width;


       public Rectangle()
        {
           this.setLength(1);
           this.setWidth(1);
        }

       public Rectangle(double l, double w)
         {
            this.setLength(l);
            this.setWidth(w);
         }

        public void setLength(double l)
         {
             if(l <= 0 || l >=20)
              {
                System.out.println("Sorry");
               throw new IllegalArgumentException();

            
               }
           else
           {
             this.length = l;
            }
        }

         public void setWidth(double w)
          {
             if(w <= 0 || w>=20)
               {
               System.out.println("Sorry");
               throw new IllegalArgumentException();
                
            
               }

              else
             {

             this.width = w;
             }
          }

         public double getWidth()
          {
              return this.width;
          }
      
         public double getLength()
           {
               return this.length;
           }

          public double perimeter()
           {
              return 2 * ( this.getWidth() + this.getLength() );
            }

          public double area()
           {
              return this.getWidth() * getLength();
           }

      }// end of class

         

File: OOPandD/MalikOOP/Malik - Inheritance - Shapes - dropped/rectangle.cpp
#include <iostream>
using namespace std; 
#include "rectangle.h"

void rectangle::setDimenstion(double l, double w)
{
    if(l>=0) rectangle::length = l;
    else rectangle::length = 0;
    if(w>=0)rectangle::width = w;
    else rectangle::width = 0;
};

double rectangle::get_length()
{
    return rectangle::length;
};

double rectangle::get_width()
{
    return rectangle::width;
};

void rectangle::set_length(double l)
{
    rectangle::length = l;
};

void rectangle::set_width(double w)
{
    rectangle::width = w;
};

double rectangle::area()
{
    return rectangle::length * rectangle::width;
};

double rectangle::perimeter()
{
    return 2*(rectangle::length + rectangle::width);
};

void rectangle::print()
{
    cout << "the length" << rectangle::length << endl;
    cout << "the width" << rectangle::width << endl;
}

rectangle::rectangle()      //default constructor 
{
    length = width = 0;
};

rectangle::rectangle(double l,double w) //constructor with parameters
{
    length = l;
    width = w;
};




